packages/backend/src/core/discovery/discover.ts

```diff
@@ -1,16 +1,19 @@
 import { EthereumAddress } from '@l2beat/shared'
 
-import { AnalyzedData, analyzeItem } from './analyzeItem'
+import { AddressAnalyzer, Analysis } from './analysis/AddressAnalyzer'
 import { DiscoveryConfig } from './config/DiscoveryConfig'
+import { HandlerExecutor } from './handlers/HandlerExecutor'
 import { DiscoveryProvider } from './provider/DiscoveryProvider'
+import { ProxyDetector } from './proxies/ProxyDetector'
+import { SourceCodeService } from './source/SourceCodeService'
 import { DiscoveryLogger } from './utils/DiscoveryLogger'
 
 export async function discover(
   provider: DiscoveryProvider,
   config: DiscoveryConfig,
   logger: DiscoveryLogger,
 ) {
-  const resolved = new Map<EthereumAddress, AnalyzedData>()
+  const resolved = new Map<EthereumAddress, Analysis>()
```

Comment:
This introduces a change to `discovered.json` (added key `type: "Contract"` to all contracts). Honestly I think that this is probably a good change (fewer interfaces, more logical data path through code), but adding this requires from us either updating all discovered.json OR agreeing that we'll see our bot to go mad. 

BTW, I wouldn't have spotted that if I haven't started the bot locally. Try adding that to your workflow, I can help you out in setting it up.

Reply:
I'm leaning towards `yarn discover --all --preserve-block-number`. Looks like an easy fix, but this script will take some time :/.

Reply:
I even thought about needing to remove that key, but later forgot. Good catch!